lib/gollum/views/pages.rb

```diff
@@ -37,34 +37,34 @@ def delete_file(url)
       def files_folders
         if has_results
           folders = {}
-          page_files = {}
+          files = {}
 
           # 1012: Folders and Pages need to be separated
-          @results.each do |page|
-            page_path = page.path
-            page_path = page_path.sub(/^#{Regexp.escape(@path)}\//, '') unless @path.nil?
+          @results.each do |result|
+            result_path = result.path
+            result_path = result_path.sub(/^#{Regexp.escape(@path)}\//, '') unless @path.nil?
             
 
-            if page_path.include?('/')
-              folder      = page_path.split('/').first
+            if result_path.include?('/')
+              # result contains a folder
+              folder      = result_path.split('/').first
               folder_path = @path ? "#{@path}/#{folder}" : folder
               folder_link = %{<li><a href="#{@base_url}/pages/#{folder_path}/" class="folder">#{folder}</a></li>}
 
               folders[folder] = folder_link unless folders.key?(folder)
-            elsif page_path != ".gitkeep"
-              if defined? page.format
-                url = "#{@base_url}/#{page.escaped_url_path}"
-              else
-                url = "#{@base_url}/#{page.escaped_url_path}#{page.name}"
-              end
-              page_link = %{<li><a href="#{url}" class="file">#{page.name}</a>#{delete_file(url) if @allow_editing}</li>}
-              page_files[page.name] = page_link
+            elsif result_path != ".gitkeep"
```

Comment:
Nice catch of the `.gitkeep` edge case 👍

Reply:
Not mine, but yeah. :)